### ISSUE TITLE
DOC: document the minimum required hardware

### ISSUE THREAD

**Issue by mattip** [numpy/numpy#17713]
In pr gh-17102 we make a hard requirement on the "baseline profile" in order to use the binary package compiled with default settings. We should document that the minimum baseline is a hard requirement for using the NumPy source code as-is, otherwise compilation is required with a lower `--cpu-baseline`. Our installation documentation lives at numpy/numpy.org. Perhaps a sentence on this page https://numpy.org/install/ about minimum hardware requirements would be appropriate, with a comment in the numpy source code at the point where [we define the `min` baseline](https://github.com/numpy/numpy/blob/8829b807a841911ce18e79b308fee9fb92fb91b6/numpy/distutils/ccompiler_opt.py#L214) that links to https://numpy.org/install/. Note also that OpenBLAS has a [similar requirement](https://github.com/MacPython/openblas-libs/blob/master/travis-ci/build_steps.sh#L88)

We might want to revisit the `RuntimeError` that could be raised when trying to import `_multiarray_ufunc` due to a non-conformant hardware platform, and make sure the [error message](https://github.com/numpy/numpy/blob/459991afcddfce6fbbae3fa83467fe607f2de7d0/numpy/core/src/common/npy_cpu_features.c.src#L164) provides enough information for a naive user to proceed. In many cases, they will be unable to compile NumPy on such weak hardware.

**Comment by mattip** [comment]
Marking this with the 1.20 milestone to be sure we resolve this before the release.

**Comment by charris** [comment]
Any reason to think this will be ready in time?

**Comment by mattip** [comment]
No. Pushing this off. I think the minimum is not a real problem: there should be very few machines out there without this minimum feature set:
```
conf_min_features = dict(
        x86 = "SSE SSE2",
        x64 = "SSE SSE2 SSE3",
        ppc64 = '', # play it safe
        ppc64le = "VSX VSX2",
        armhf = '', # play it safe
        aarch64 = "NEON NEON_FP16 NEON_VFPV4 ASIMD"
    )
```
